src/backend/Core/Services/RoleService.cs

```diff
@@ -0,0 +1,137 @@
+﻿using Core.Database;
+using Core.Database.Enums;
+using Core.Database.Models.Auth;
+using Core.Exceptions;
+using Core.Setup.Auth0;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+
+namespace Core.Services
+{
+    public class RoleService
+    {
+        private readonly ICurrentUserService _currentUserService;
+        private readonly AppDbContext _appDbContext;
+        private readonly ILogger<RoleService> _logger;
+        private readonly GetCurrentUserService _getCurrentUserService;
+
+        public RoleService(ILogger<RoleService> logger, AppDbContext appDbContext, ICurrentUserService currentUserService, GetCurrentUserService getCurrentUserService)
+        {
+            _appDbContext = appDbContext;
+            _logger = logger;
+            _currentUserService = currentUserService;
+            _getCurrentUserService = getCurrentUserService;
+        }
+
+        public async Task<List<RoleName>> GetRolesAsync(long? userId = null)
+        {
+            if (userId == null)
+            {
+                var roleNames = await GetCurrentUserRolesRoleNames();
+                return roleNames;
+            }
+
+            var roleNames2 = await GetUserRolesRoleNamesFromUserWithId(userId);
+
+            return roleNames2;
+        }
+
+        public async Task<bool> HasRole(RoleName role, long? userId = null)
+        {
+            var listOfUserRolesRoleNames = await GetRolesAsync(userId);
+
+            var userHasGivenRole = listOfUserRolesRoleNames.Any(c => c == role);
+
+            return userHasGivenRole;
+        }
+
+        public async Task AssertRole(RoleName roleName, long? userId = null)
```

Comment:
Wejdź do clickupa, przeczytaj co ta funkcja ma robić i porównaj ze swoją implementacją.

Reply:
Jeśli wg Ciebie nazwa jest nieadekwatna, to można zmienić np na `ThrowIfNoRole()` czy cos